services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -316,8 +323,12 @@ public boolean checkTokenValidForIntrospection(KeycloakSession session, RealmMod
                     clientSession.setTimestamp(currentTime);
                 }
             }
+            eventBuilder.detail("Token valid for introspection", String.valueOf(valid));
+            
         }
 
+        if (!valid)
+            logger.warn("Introspection access token: Failed to valid access token for introspection");
```

Comment:
Is it possible to debug instead of warn and possibly add more context data?

Reply:
> Is it possible to debug instead of warn and possibly add more context data?

I have created appropriate debug logs and event deails for all possible cases of valid = false.